mcs/class/corlib/corefx/Interop.GetRandomBytes.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+
+internal partial class Interop
+{
+	static class MonoGetRandomBytesFallback
+	{
+		static object _rngAccess = new object ();
+		static RandomNumberGenerator _rng;
+
+		internal static void Test (byte[] buffer)
+		{
+			lock (_rngAccess) {
+				if (_rng == null)
+					_rng = new RNGCryptoServiceProvider ();
+				_rng.GetBytes (buffer);
+			}
+		}
+	}
+
+	internal static unsafe void GetRandomBytes (byte* buffer, int length)
+	{
+		var bytes = new byte[length];
```

Comment:
If we are taking this approach, we should not have to allocate here, but instead pass the byte* to the runtime so we populate it without copying.